lib/hipe/test/basic_SUITE_data/basic_tuples.erl

```diff
@@ -73,6 +75,16 @@ get_elements([Element|Rest], Tuple, Pos) ->
 get_elements([], _Tuple, _Pos) ->
   ok.
 
+squares() ->
+  {1*1,   2*2,   3*3,   4*4,   5*5,   6*6,   7*7,   8*8,   9*9,   10*10,
+   11*11, 12*12, 13*13, 14*14, 15*15, 16*16, 17*17, 18*18, 19*19, 20*20}.
+
+test_square() ->
+  %% A bit of gymnastics to create a variable with a 1..19 range
```

Comment:
You're right. It writes `v6 (-9..9) := 'rem'(v4 (any()), 9) (primop)`, but should be `-8..8`. Further investigation show that `erl_bif_types:arith_rem(neg_inf,pos_inf,9,9) = {-8,8}` produces the right answer, but `hipe_icode_range:range_rem({range,{neg_inf,pos_inf},false},{range,{9,9},false}) = {range,{-9,9},false}` does not. Maybe we should just have `hipe_icode_range` defer to `erl_bif_types` rather than reimplementing everything (plus bugs).